src/coreclr/jit/valuenum.cpp

```diff
@@ -2675,7 +2675,7 @@ ValueNum ValueNumStore::EvalFuncForConstantArgs(var_types typ, VNFunc func, Valu
             {
                 assert((typ == TYP_BYREF) || (typ == TYP_LONG));
```

Comment:
can the assert be changed to `assert((typ == TYP_BYREF) || (typ == TYP_I_IMPL));` so we are sure we don't cast LONG to 32-bit on x86/arm32?

Reply:
OK